[tm-bench] runtime error: slice bounds out of range

tendermint version: 0.23.1-9e940b95

Steps to reproduce:

[vagrant@dev tm-bench]$tm-bench -T 10 -r 10 -s 39 localhost:26657
panic: runtime error: slice bounds out of range

goroutine 13 [running]:
main.generateTx(0x0, 0x0, 0x27, 0xf7ec95a0bcd06a05, 0x11c84eadb4a4a45b, 0x20, 0xc4200fcb00, 0xc420705d58)
	/home/vagrant/gocode/src/github.com/tendermint/tendermint/tools/tm-bench/transacter.go:266 +0x263
main.(*transacter).sendLoop(0xc42055ea00, 0x0)
	/home/vagrant/gocode/src/github.com/tendermint/tendermint/tools/tm-bench/transacter.go:165 +0x2fb
created by main.(*transacter).Start
	/home/vagrant/gocode/src/github.com/tendermint/tendermint/tools/tm-bench/transacter.go:82 +0x186

--- a/tools/tm-bench/main.go
+++ b/tools/tm-bench/main.go
@@ -25,7 +25,7 @@ func main() {
 	flagSet.IntVar(&connections, "c", 1, "Connections to keep open per endpoint")
 	flagSet.IntVar(&durationInt, "T", 10, "Exit after the specified amount of time in seconds")
 	flagSet.IntVar(&txsRate, "r", 1000, "Txs per second to send in a connection")
-	flagSet.IntVar(&txSize, "s", 250, "The size of a transaction in bytes.")
+	flagSet.IntVar(&txSize, "s", 250, "The size of a transaction in bytes, must be greater than or equal to 40.")
 	flagSet.StringVar(&outputFormat, "output-format", "plain", "Output format: plain or json")
 	flagSet.StringVar(&broadcastTxMethod, "broadcast-tx-method", "async", "Broadcast method: async (no guarantees; fastest), sync (ensures tx is checked) or commit (ensures tx is checked and committed; slowest)")
 	flagSet.BoolVar(&verbose, "v", false, "Verbose output")
@@ -66,6 +66,14 @@ Examples:
 		logger = log.NewTMLoggerWithColorFn(log.NewSyncWriter(os.Stdout), colorFn)
 
 		fmt.Printf("Running %ds test @ %s\n", durationInt, flagSet.Arg(0))
+	}
+
+	if txSize < 40 {
+		fmt.Fprintln(
+			os.Stderr,
+			"The size of a transaction must be greater than or equal to 40.",
+		)
+		os.Exit(1)
 	}
 
 	if broadcastTxMethod != "async" &&

--- a/tools/tm-bench/transacter_test.go
+++ b/tools/tm-bench/transacter_test.go
@@ -22,7 +22,7 @@ func TestGenerateTxUpdateTxConsistentency(t *testing.T) {
 		hostname         string
 		numTxsToTest     int
 	}{
-		{0, 0, 50, "localhost:26657", 1000},
+		{0, 0, 40, "localhost:26657", 1000},
 		{70, 300, 10000, "localhost:26657", 1000},
 		{0, 50, 100000, "localhost:26657", 1000},
 	}

